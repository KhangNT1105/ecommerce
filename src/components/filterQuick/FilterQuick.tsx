import React from 'react';
import Styles from './FilterQuick.module.scss';

interface FilterItem {
  field: string;
  value: string;
  chosen?: boolean;
  title?: string;
}

interface FilterQuickProps {
  filterList: FilterItem[];
  onChooseFilter?: (field: string, value: string) => void;
  className?: string;
}

const FilterQuick: React.FC<FilterQuickProps> = ({ filterList, onChooseFilter, className }) => {
  const handleOnClick = (field: string, value: string) => () => {
    onChooseFilter && onChooseFilter(field, value);
  };

  return (
    <div className={`${Styles['filter-quick']} ${className}`}>
      <div className="filter-quick-title">Quick Filters</div>
      {filterList.map(({ field, value, chosen, title }, index) => (
        <div className={chosen ? 'filter-quick-chosen' : ''} key={index} onClick={handleOnClick(field, value)}>
          {title || value}
        </div>
      ))}
    </div>
  );
};

export default FilterQuick;
