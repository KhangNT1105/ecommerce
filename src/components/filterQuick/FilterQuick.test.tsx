import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import FilterQuick from './FilterQuick';

describe('<FilterQuick />', () => {
  const defaultProps = {
    filterList: [
      {
        field: 'status',
        value: 'Active'
      },
      {
        field: 'type',
        value: 'Admin'
      },
      {
        field: 'type',
        value: 'User'
      },
      {
        field: 'status',
        value: 'inactive',
        title: 'Inactive'
      }
    ]
  };
  const wrapper = renderer.create(<FilterQuick {...defaultProps} />);

  test('render snapshot correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FilterQuick event', () => {
  const defaultProps = {
    filterList: [
      {
        field: 'status',
        value: 'Active',
        chosen: true
      },
      {
        field: 'type',
        value: 'Admin'
      },
      {
        field: 'type',
        value: 'User'
      }
    ],
    onChooseFilter: jest.fn()
  };
  const instance = shallow(<FilterQuick {...defaultProps} />);
  instance.find('.filter-quick-chosen').first().simulate('click');

  expect(defaultProps.onChooseFilter).toHaveBeenCalled();
});
