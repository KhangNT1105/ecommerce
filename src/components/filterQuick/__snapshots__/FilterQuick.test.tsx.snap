// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterQuick /> render snapshot correctly 1`] = `
<div
  className="filter-quick undefined"
>
  <div
    className="filter-quick-title"
  >
    Quick Filters
  </div>
  <div
    className=""
    onClick={[Function]}
  >
    Active
  </div>
  <div
    className=""
    onClick={[Function]}
  >
    Admin
  </div>
  <div
    className=""
    onClick={[Function]}
  >
    User
  </div>
  <div
    className=""
    onClick={[Function]}
  >
    Inactive
  </div>
</div>
`;
